src/Lean/Elab/Tactic/LibrarySearch.lean

```diff
@@ -0,0 +1,86 @@
+/-
+Copyright (c) 2021-2024 Gabriel Ebner and Lean FRO. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gabriel Ebner, Joe Hendrix, Scott Morrison
+-/
+import Lean.Meta.Tactic.LibrarySearch
+import Lean.Meta.Tactic.TryThis
+import Lean.Elab.Tactic.BuiltinTactic
+
+namespace Lean.Elab.LibrarySearch
+
+open Lean Meta LibrarySearch
+open Elab Tactic TryThis
+
+
+-- TODO: implement the additional options for `library_search` from Lean 3,
+-- in particular including additional lemmas
+-- with `std_exact? [X, Y, Z]` or `std_exact? with attr`.
+
+-- For now we only implement the basic functionality.
+-- The full syntax is recognized, but will produce a "Tactic has not been implemented" error.
+
+/-- Implementation of the `exact?` tactic. -/
```

Comment:
See suggestions on syntax for a docstring that's a bit more useful.

Here, it'd be good to see docs for the meaning of each argument.